Files/Views/LayoutModes/GenericFileBrowser.xaml.cs

```diff
@@ -262,34 +259,6 @@ private void AllView_PreparingCellForEdit(object sender, DataGridPreparingCellFo
                 return;
             }
 
-            // Only cancel if this event was triggered by a tap
-            // Do not cancel when user presses F2 or context menu
-            if (e.EditingEventArgs is TappedRoutedEventArgs)
-            {
-                if (AppSettings.OpenItemsWithOneclick)
```

Comment:
This is still neded I think, try testing with `OpenItemsWithOneclick` option enabled

Reply:
Sorry

Reply:
@gave92 - before I fix this, I never worked single-click mode. Is there an option at all to rename based on mouse interaction?

Reply:
Not in single-click mode, rename is only with "F2" or mouse right click

Reply:
@gave92 - in this case I am not sure why we need this logic in cell editing as it should be already handled in ItemClick but I guess it was added for reason.

Reply:
The DataGrid would put the cell in rename mode, can you point me to the ItemClick you are referring to? It should be only in GridView not Listview

Reply:
I was actually referring to `AllView_ItemPress`. 
After few minutes of playing I was able to get to PrepairingEdit in one-click mode.. but it might originate from some other defect...

Reply:
@gave92 - I think there are several issues here:
1. `AllView_ItemPress` should mark event as handled if it decides to invoke ItemClick, so it won't get routed to the Grid  (to avoid editing from being invoked).
2. There RectangleSelection creates another issue - it unregisters `All_ViewItemPress` and.. it is too sensitive and is started even if from my perspective I didn't move the mouse after clicking.  So sometimes when I click, the rectangle selection kicks in, unregisters AllView_ItemPressed and nothing happens upon click.

Reply:
I can return the protection, to edit - but it should never be invoked in normal conditions.

Reply:
> I think there are several issues here

Mouse/Pointer/Keyboard event handling is a mess, yeah
1- Ok, if it prevents AllView_PreparingCellForEdit from firing in case of single click mode, then we can remove the whole code section as you did
2- Selection rectangle's SelectionStarted should be only called if the mouse was moved at least a bit

Reply:
> In single-click mode rename is only with "F2" or mouse right click (context menu) <- confirm @yaichenbaum?

That's correct.

Reply:
@gave92 - I returned the protection and I am changing the sensitivity of selection in a separate PR

Reply:
@Don-Vito @gave92 Looking at the [docs](https://docs.microsoft.com/en-us/windows/communitytoolkit/controls/datagrid_guidance/editing_inputvalidation#editing-methods-and-events), it is actually unnecessary to handle the `PreparingCellForEdit` event. We should evaluate whether handling the `BeginningEdit` event will suffice for our special processing since the `BeginningEdit` event does occur **before** a cell or row enters editing mode instead of when a cell enters the editing mode.

Basically since `PreparingCellForEdit` is technically a method as per the docs, it may be bad practice to override it with our own logic.